src/Controllers/ElementalAreaController.php

```diff
@@ -158,6 +355,36 @@ public static function removeNamespacesFromFields(array $data, $elementID)
         return $output;
     }
 
+    private function getNewTitle(string $title = ''): ?string
+    {
+        $hasCopyPattern = '/^.*(\scopy($|\s[0-9]+$))/';
+        $hasNumPattern = '/^.*(\s[0-9]+$)/';
+        $parts = [];
+        // does $title end with 'copy' (ignoring numbers for now)?
+        if (preg_match($hasCopyPattern ?? '', $title ?? '', $parts)) {
+            $copy = $parts[1];
+            // does $title end with numbers?
+            if (preg_match($hasNumPattern ?? '', $copy ?? '', $parts)) {
+                $num = trim($parts[1] ?? '');
+                $len = strlen($num ?? '');
+                $inc = (int)$num + 1;
+                return substr($title ?? '', 0, -$len) . "$inc";
+            } else {
+                return $title . ' 2';
+            }
+        } else {
+            return $title . ' copy';
+        }
+    }
+
+    // TODO: upgrade to AssesAdmin:: get-post-data
+    private function getPostData()
+    {
+        $request = $this->getRequest();
+        $postData = json_decode($request->getBody() ?? '', true);
+        return $postData;
+    }
```

Comment:
I assume the TODO means you intended to centralise that method - I agree it seems like it should be centralised. This is the third time I've seen this method (or one very similar) being added I think.

Reply:
Am now using the new method on LeftAndMain